crates/services/importer/src/importer.rs

```diff
@@ -0,0 +1,241 @@
+use crate::{
+    ports::{
+        BlockVerifier,
+        Executor,
+        ExecutorDatabase,
+        ImporterDatabase,
+    },
+    Config,
+};
+use anyhow::anyhow;
+use fuel_core_storage::{
+    transactional::StorageTransaction,
+    IsNotFound,
+};
+use fuel_core_types::{
+    blockchain::{
+        consensus::{
+            Consensus,
+            Sealed,
+        },
+        primitives::BlockHeight,
+        SealedBlock,
+    },
+    fuel_vm::crypto::ephemeral_merkle_root,
+    services::{
+        block_importer::{
+            ImportResult,
+            UncommittedResult,
+        },
+        executor::{
+            ExecutionBlock,
+            ExecutionResult,
+        },
+        Uncommitted,
+    },
+};
+use std::sync::Arc;
+use tokio::sync::broadcast;
+
+pub struct Importer<D, E, V> {
+    database: D,
+    executor: E,
+    verifier: V,
+    broadcast: broadcast::Sender<Arc<ImportResult>>,
+    guard: tokio::sync::Mutex<()>,
```

Comment:
There's no need to lock the execution. The call to `execute_and_commit` should have a semaphore with a single permit so that it can only be called once.